controller/settings.go

```diff
@@ -47,3 +47,35 @@ func (o OnConnectSettingsHandler) RouterConnected(r *network.Router) {
 		}).Info("no on connect settings to send")
 	}
 }
+
+type OnConnectCtrlAddressesUpdateHandler struct {
```

Comment:
I see how the callback works, but I think it would be cleared if you just had the network in here and had the GetAddresses as a method on the handler